### PR TITLE
Cell chargers are now buildable and (upgradable!) machines.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
add: Cell chargers can now be built and upgraded with capacitors!
fix: Fixed empty subtype batteries not updating icons
/:cl:

[why]: This is a needed change since batteries are now printed empty.
@Cyberboss i did it bae

### CODE REVIEW

**Comment by DaedalusGame** [comment]
Can we make these handheld for gods sake

**Comment by DaxDupont** [comment]
Not without a large refactor, besides these are supposed to wired or something.

**Comment by DaxDupont** [comment]
You can drag them around if you unwrench them tho

**Comment by DaedalusGame** [comment]
ughhhhh using 10 sheets of metal to build a tiny recharger that looks like it'd fit in my pocket

**Comment by DaxDupont** [comment]
oh wait i forgot to change the material amount

**Comment by DaxDupont** [comment]
Actually this doesn't consume metal aside from the 5 for the frame

**Comment by SgtxRootbeer** [comment]
We have inducers for hand-held use

**Comment by DaedalusGame** [comment]
inducers use cells

**Comment by DaxDupont** [comment]
they can also charge cells

**Comment by DaedalusGame** [comment]
i don't know what kind of argument you're trying to make!

**Comment by DaxDupont** [comment]
Well if you want to hand held charge then use inducers, if you want to draw from the APC, that's what cell chargers are for.

**Comment by GunHog** [comment]
### **THANK YOU SO MUCH!**

**Comment by Kromgar** [comment]
speedmerge pls

File: code/game/machinery/cell_charger.dm [hunk 1]
```diff
@@ -52,17 +54,19 @@
...
-
-		anchored = !anchored
-		to_chat(user, "<span class='notice'>You [anchored ? "attach" : "detach"] [src] [anchored ? "to" : "from"] the ground</span>")
-		playsound(src.loc, W.usesound, 75, 1)
-	else
 		return ..()
 
+/obj/machinery/cell_charger/Destroy()
+	if(charging)
+		charging.forceMove(drop_location())
```
Comment (Cyberboss) [inline code/game/machinery/cell_charger.dm]: It should just QDEL_NULL it if it's supposed to be inside it

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/cell_charger.dm (hunk 1 above)
Comment (DaxDupont) [reply code/game/machinery/cell_charger.dm]: Eh if someone beats it till it breaks the battery should fall out of it though? I mean it's ontop of it.

Comment (DaxDupont) [reply code/game/machinery/cell_charger.dm]: I can make it QDELL_NULL but I am not sure it should eat the battery really

Comment (DaedalusGame) [reply code/game/machinery/cell_charger.dm]: Deconstruct should return the battery instead if I understand correctly

Comment (DaxDupont) [reply code/game/machinery/cell_charger.dm]: Let me double check real quick

Comment (Cyberboss) [reply code/game/machinery/cell_charger.dm]: deconstuct can move it, Destroy should delete it

File: code/game/machinery/cell_charger.dm [hunk 2]
```diff
@@ -52,17 +54,24 @@
...
 		return ..()
 
+/obj/machinery/cell_charger/deconstruct()
+	if(charging)
+		charging.forceMove(drop_location())
+	return ..()
+
+/obj/machinery/cell_charger/Destroy()
+	if(charging)
+		QDEL_NULL(charging)
```
Comment (Cyberboss) [inline code/game/machinery/cell_charger.dm]: remove the if

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by DaxDupont** [comment]
oh great fuck

**Comment by DaxDupont** [comment]
fixing it

**Comment by Cyberboss** [comment]
gg

**Comment by DaxDupont** [comment]
@Cyberboss should fix it

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**